lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp

```diff
@@ -142,9 +188,26 @@ splitAggregateLoad(LoadInst *loadInst, CanonicalizeInstruction &pass) {
 
   // Add load projections to a projection list.
   llvm::SmallVector<ProjInstPair, 8> projections;
+  llvm::SmallVector<BeginBorrowInst *, 8> borrows;
+  llvm::SmallVector<DestroyValueInst *, 8> destroys;
   for (auto *use : getNonDebugUses(loadInst)) {
     auto *user = use->getUser();
-
+    if (needsBorrow) {
+      if (auto *borrow = dyn_cast<BeginBorrowInst>(user)) {
+        // The transformation below also assumes a single borrow use.
+        auto *borrowedOper = borrow->getSingleNonEndingUse();
+        if (!borrowedOper)
+          return nextII;
+
+        borrows.push_back(borrow);
+        user = borrowedOper->getUser();
+
+      } else if (auto *destroy = dyn_cast<DestroyValueInst>(user)) {
+        destroys.push_back(destroy);
+        continue;
+      } else
+        return nextII;
```

Comment:
Can you not do an else return here? That is against LLVM style. I think you could re-organize this code to eliminate it like so:

```
if (needsBorrow) {
  if (auto *destroy = dyn_cast<DestroyValueInst>(user)) { ... }
  auto *borrow = dyn_cast<BeginBorrowInst>(user);
  if (!borrow)
    return nextII;
  ...
}
```

Reply:
OK. I originally wrote it that way. I put it back.